packages/addresses/src/address.ts

```diff
@@ -53,29 +52,25 @@ export function isAddress(putativeAddress: string): putativeAddress is Address<t
 }
 
 export function assertIsAddress(putativeAddress: string): asserts putativeAddress is Address<typeof putativeAddress> {
-    try {
-        // Fast-path; see if the input string is of an acceptable length.
-        if (
-            // Lowest address (32 bytes of zeroes)
-            putativeAddress.length < 32 ||
-            // Highest address (32 bytes of 255)
-            putativeAddress.length > 44
-        ) {
-            throw new SolanaError(SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_LENGTH, {
-                actualLength: putativeAddress.length,
-            });
-        }
-        // Slow-path; actually attempt to decode the input string.
-        const base58Encoder = getMemoizedBase58Encoder();
-        const bytes = base58Encoder.encode(putativeAddress);
-        const numBytes = bytes.byteLength;
-        if (numBytes !== 32) {
-            throw new SolanaError(SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_BYTE_LENGTH, {
-                actualLength: numBytes,
-            });
-        }
-    } catch (e) {
-        throw new SolanaError(SOLANA_ERROR__NOT_A_BASE58_ENCODED_ADDRESS, { putativeAddress });
+    // Fast-path; see if the input string is of an acceptable length.
+    if (
+        // Lowest address (32 bytes of zeroes)
+        putativeAddress.length < 32 ||
+        // Highest address (32 bytes of 255)
+        putativeAddress.length > 44
+    ) {
+        throw new SolanaError(SOLANA_ERROR__ADDRESS_STRING_LENGTH_OUT_OF_RANGE, {
+            actualLength: putativeAddress.length,
+        });
+    }
+    // Slow-path; actually attempt to decode the input string.
+    const base58Encoder = getMemoizedBase58Encoder();
+    const bytes = base58Encoder.encode(putativeAddress);
+    const numBytes = bytes.byteLength;
+    if (numBytes !== 32) {
+        throw new SolanaError(SOLANA_ERROR__ADDRESS_BYTE_LENGTH_OUT_OF_RANGE, {
+            actualLength: numBytes,
+        });
     }
```

Comment:
@buffalojoec, I missed this in #2187. This _used_ to aim to create an ‘all up’ error that the address is bad, with the inner error as the `cause`. Now that we have codes and separate errors for each cause, I don't think we need this.

Also, #2187 buried the sub errors so that they'd never reach the caller.

Reply:
Ahh, I see. Nice catch! So can't `NOT_A_BASE_58_ADDRESS` come out then?

Reply:
That one would have bubbled up, yes, but the others would never have seen the light of day.

I think in other PRs we consolidated on throwing granular errors, rather than one error with a `cause`.